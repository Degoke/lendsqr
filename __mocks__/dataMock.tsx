import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
