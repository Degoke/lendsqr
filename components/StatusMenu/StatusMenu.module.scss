@use '../../styles/variables' as *;

.popover {
  width: 160px;
  left: 0;
  top: 0;
  margin-left: -5rem;

  &__menu {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 1rem;

    a {
      text-decoration: none;
      font-weight: inherit;
    }

    &_button {
      border: none;
      background: inherit;
      font-weight: 500;
      font-size: 14px;
      color: $text-primary;

      display: flex;
      align-items: center;
      gap: 0.5rem;
    }

    &_item {
      padding: 0.5rem 0;
    }
  }
}
