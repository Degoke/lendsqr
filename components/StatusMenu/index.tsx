import Image, { StaticImageData } from 'next/image'
import Link from 'next/link'
import { selectCurrentStatusMenu } from '../../features/helpers/helperSlice'
import { useAppSelector } from '../../utils/hooks/useRedux'
import Popover from '../Popover'
import styles from './StatusMenu.module.scss'

export type StatusMenuItem = {
  image: StaticImageData
  text: string
  isLink: boolean
  onClick?: () => void
}

type Props = {
  items: StatusMenuItem[]
  id: string
}

export default function StatusMenu({ items, id }: Props) {
  const currentStatusMenu = useAppSelector(selectCurrentStatusMenu)
  return (
    <Popover isOpen={currentStatusMenu === id} className={styles.popover}>
      <ul className={styles.popover__menu}>
        {items?.map((item) => {
          if (item.isLink) {
            return (
              <li key={item.text} className={styles.popover__menu_item}>
                <Link
                  href={`/users/${id}`}
                  className={styles.popover__menu_button}
                >
                  <Image src={item.image} alt={item.text} />
                  {item.text}{' '}
                </Link>
              </li>
            )
          } else {
            return (
              <li key={item.text} className={styles.popover__menu_item}>
                <button
                  className={styles.popover__menu_button}
                  onClick={item.onClick}
                >
                  <Image src={item.image} alt={item.text} />
                  {item.text}
                </button>
              </li>
            )
          }
        })}
      </ul>
    </Popover>
  )
}
