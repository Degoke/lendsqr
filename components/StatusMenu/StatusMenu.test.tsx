import { render, screen } from '@testing-library/react'
import StatusMenu from './index'
import { mockCard, mockImage } from '../../__mocks__/dataMock'
import { mockProvider } from '../../jest.setup'

describe('StatusMenu tests', () => {
  it('renders StatusMenu component unchanged', () => {
    const { container } = render(
      mockProvider(<StatusMenu items={[]} id={''} />)
    )
    expect(container).toMatchSnapshot()
  })
})
