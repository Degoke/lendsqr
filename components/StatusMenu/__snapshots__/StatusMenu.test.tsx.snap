// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusMenu tests renders StatusMenu component unchanged 1`] = `
<div>
  <div
    class="popover popover"
  >
    <div
      class="paper undefined"
    >
      <ul
        class="popover__menu"
      />
    </div>
  </div>
</div>
`;
